test/components/map.spec.js

```diff
@@ -38,38 +100,65 @@ test('InteractiveMap#named export', t => {
   t.end();
 });
 
-test('InteractiveMap#call onLoad when provided', t => {
-  function onLoad(...args) {
-    t.is(args.length, 0, 'onLoad does not expose the map object.');
-    t.end();
-  }
+TEST_CASES.forEach(testCase => {
+  test(`InteractiveMap#load ${testCase.title}`, t => {
+    let result = null;
+    let timer = null;
 
-  const props = Object.assign({}, defaultProps, {onLoad});
+    function end() {
+      const warning = result.root.findAllByType('h3');
+      if (testCase.shouldLoad) {
+        t.notOk(warning.length, 'shouldn\'t show warning');
+      } else {
+        t.ok(warning.length, 'should show warning');
+      }
 
-  const map = createElement(InteractiveMap, props);
+      result.unmount();
+      t.end();
+    }
 
-  const result = ReactTestRenderer.create(map);
+    function onLoad() {
+      if (testCase.shouldLoad) {
+        t.pass('onLoad is called');
+      } else {
+        t.fail('onLoad should not be called');
+      }
+      clearTimeout(timer);
+      end();
```

Comment:
Prefer calling `t.end()` at top level rather than inside a nested method.

Also maybe rename `end` to `check` or similar?